### ISSUE TITLE
Move typeOf methods to `bem-entity-name` package

### ISSUE THREAD

**Issue by blond** [bem-sdk/bem-naming#98]
* [x] add `typeOf` method to `BemEntityName`: We can use [type](https://github.com/bem-sdk/bem-entity-name#type) field in `BemEntityName` instance.
* [x] remove `typeOf` method.

**Comment by blond** [comment]
Related https://github.com/bem-sdk/bem-entity/issues/25

**Comment by blond** [comment]
let's  remove `validate` method and will move it to `bem-entity-part`.

If we want validate string we can use `parse` method:

```js
const bemNaming = require('bem-naming');

bemNaming.parse('9#$%85#$'); // undefined
```

If we want validate object or `BemEntityPart` instance we can use `typeOf` method: 

```js
const BemEntityPart = require('bem-entity-part');

BemEntityPart.typeOf({ elem: 'bla' }); // undefined
```

**Comment by blond** [comment]
Fixed in #145
